### PR TITLE
2205 Add IIdentifiable<T> and ICoded<T> to POCOs

### PR DESCRIPTION
**Pull request by ewoutkramer**
This provides a consistent interface to retrieve (and set) identifiers and codes on identifiable and coded poco's .

* Identifiable POCOs are poco's that have an element called "Identifier". 
* Coded POCO's are poco's that have a "primary code path" according to the FHIR binding for CQL. Many times, these are POCO's with a "Code" element, but it could also be called differently.  Because the CQL binding is only specified for R4, these interfaces are only used on R4 POCO's (and POCO's in common).

### CODE REVIEW

**Comment by ewoutkramer** [comment]
This PR contains the same functionality as #2496, but uses codegen directly instead of post-processing of the source files after codegen.

**Comment by ewoutkramer** [comment]
Fixes #2205

File: src/Hl7.Fhir.Base/Model/Generated/Coding.cs [hunk 1]
```diff
@@ -121,7 +121,6 @@ public string Version
     /// Symbol in syntax defined by the system
     /// </summary>
     [FhirElement("code", InSummary=true, Order=50)]
-    [CqlElement(IsPrimaryCodePath = true)]
```
Comment (marcovisserFurore) [inline src/Hl7.Fhir.Base/Model/Generated/Coding.cs:124]: Should this class not implementing `ICoded` as well?

File: src/Hl7.Fhir.Base/Model/Generated/Quantity.cs [hunk 2]
```diff
@@ -225,7 +225,6 @@ public string System
     /// Coded form of the unit
     /// </summary>
     [FhirElement("code", InSummary=true, Order=70)]
-    [CqlElement(IsPrimaryCodePath = true)]
```
Comment (marcovisserFurore) [inline src/Hl7.Fhir.Base/Model/Generated/Quantity.cs:228]: Should this class not implementing `ICoded` as well?

File: src/Hl7.Fhir.Base/Model/Generated/UsageContext.cs [hunk 3]
```diff
@@ -59,7 +59,7 @@ public partial class UsageContext : Hl7.Fhir.Model.DataType
     /// Type of context being specified
     /// </summary>
     [FhirElement("code", InSummary=true, Order=30)]
-    [CqlElement(IsPrimaryCodePath=true)]
```
Comment (marcovisserFurore) [inline src/Hl7.Fhir.Base/Model/Generated/UsageContext.cs:62]: Should this class not implementing `ICoded` as well?

**Review by marcovisserFurore — CHANGES_REQUESTED** [review]
Some remarks about missing interfaces....

File: src/Hl7.Fhir.Base/Model/Generated/Coding.cs (hunk 1 above)
Comment (ewoutkramer) [reply src/Hl7.Fhir.Base/Model/Generated/Coding.cs:124]: No, ICoded<T> should only be used on resources - this is a mistake in the CQL specification and I have corrected it here.

**Review by marcovisserFurore — APPROVED** [review]

**Comment by taconaut** [comment]
Thanks 👍 , I'll check you solved this.

**Pull request merged**